owncloudApp/src/main/java/com/owncloud/android/presentation/ui/files/removefile/RemoveFilesDialogFragment.kt

```diff
@@ -0,0 +1,136 @@
+ * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
+ */
+package com.owncloud.android.presentation.ui.files.removefile
+
+import android.app.Dialog
+import android.os.Bundle
+import com.owncloud.android.R
+import com.owncloud.android.domain.files.model.OCFile
+import com.owncloud.android.presentation.ui.files.operations.FileOperation
+import com.owncloud.android.presentation.ui.files.operations.FileOperationViewModel
+import com.owncloud.android.ui.dialog.ConfirmationDialogFragment
+import com.owncloud.android.ui.dialog.ConfirmationDialogFragment.ConfirmationDialogFragmentListener
+import org.koin.androidx.viewmodel.ext.android.sharedViewModel
+import java.util.ArrayList
+
+/**
+ * Dialog requiring confirmation before removing a collection of given OCFiles.
+ *
+ * Triggers the removal according to the user response.
+ */
+class RemoveFilesDialogFragment : ConfirmationDialogFragment(), ConfirmationDialogFragmentListener {
+
+    private lateinit var targetFiles: ArrayList<OCFile>
+    private val fileOperationViewModel: FileOperationViewModel by sharedViewModel()
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = super.onCreateDialog(savedInstanceState)
+        targetFiles = requireArguments().getParcelableArrayList(ARG_TARGET_FILES) ?: arrayListOf()
+        setOnConfirmationListener(this)
+        return dialog
+    }
+
+    /**
+     * Performs the removal of the target file, both locally and in the server.
+     */
+    override fun onConfirmation(callerTag: String) {
+        fileOperationViewModel.performOperation(FileOperation.RemoveOperation(targetFiles.toList(), removeOnlyLocalCopy = false))
+        dismiss()
+    }
+
+    /**
+     * Performs the removal of the local copy of the target file
+     */
+    override fun onCancel(callerTag: String) {
+        fileOperationViewModel.performOperation(FileOperation.RemoveOperation(targetFiles.toList(), removeOnlyLocalCopy = true))
+        dismiss()
+    }
+
+    override fun onNeutral(callerTag: String) {
+        // nothing to do here
+    }
+
+    companion object {
+        private const val ARG_TARGET_FILES = "TARGET_FILES"
+
+        /**
+         * Public factory method to create new RemoveFilesDialogFragment instances.
+         *
+         * @param files           Files to remove.
+         * @return Dialog ready to show.
+         */
+        @JvmStatic
+        fun newInstance(files: ArrayList<OCFile>): RemoveFilesDialogFragment {
+            val messageStringId: Int
+            var containsFolder = false
+            var containsDown = false
+            val containsAvailableOffline = false
+            for (file in files) {
+                if (file.isFolder) {
+                    containsFolder = true
+                }
+                if (file.isAvailableLocally) {
+                    containsDown = true
+                }
+                // FIXME: 13/10/2020 : New_arch: Av.Offline
```

Comment:
Maybe is better to use a `TODO` with a `New arch` tag so you can filter in the future what's pending? 🍻

Reply:
I introduced fixmes when we migrated the database, so I can filter them to check what's missing to move to the new arch. The idea is basically the same as TODOS, so I'll keep it this way since I already started using fixmes. Anyway, I will consider your proposal for future refactors 👍